### PR TITLE
[CLI Config Parity] migrate log-file legacy flag/config to new config 

### PR DESCRIPTION
**Pull request by ashmeenkaur**
### Description
This PR is part of migration to new cobra/viper configs for CLI config parity.
Replaced usage of `log-file` flag stored in `flagstorage.LogFile` and `mountConfig.LogConfig.FilePath` to `newConfig.Logging.FilePath`.

Note: Post merging this PR, log-file flag will start taking precedence when both flag and config is set.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - Manually verified that the new flag is working
2. Unit tests - Updated
3. Integration tests - Ran via KOKORO

### CODE REVIEW

File: tools/integration_tests/log_rotation/log_rotation_test.go [hunk 1]
```diff
@@ -94,6 +94,7 @@ func TestMain(m *testing.M) {
 	// Set up directory for logs.
 	logDirPath = setup.SetUpLogDirForTestDirTests(logDirName)
 	logFilePath = path.Join(logDirPath, logFileName)
+	setup.SetLogFile(logFilePath)
```
Comment (kislaykishore) [inline tools/integration_tests/log_rotation/log_rotation_test.go:97]: Why is this change needed?

**Review by kislaykishore — APPROVED** [review]

File: tools/integration_tests/log_rotation/log_rotation_test.go (hunk 1 above)
Comment (ashmeenkaur) [reply tools/integration_tests/log_rotation/log_rotation_test.go:97]: Because log file set via flag takes precedence now

**Review by kislaykishore — APPROVED** [review]

**Review by charith87 — APPROVED** [review]
